node/tests/README.md

```diff
@@ -0,0 +1 @@
+TODO: We need a proper guide on what is going on here
```

Comment:
Maybe start with a very short description by having a simple diagram and a few bullet-points. It would also help reviewers as an intro to the PR.

Reply:
Done